src/planwise/component/auth.clj

```diff
@@ -140,24 +141,34 @@
    :token (:access-token guisso-resp)
    :refresh-token (:refresh-token guisso-resp)})
 
+(defn- fetch-token-from-guisso
+  "Performs an OAuth2 token request to Guisso using the service configuration
+  and catching possible invalid grant errors and returning nil in such cases."
+  [service params]
+  (try+
+   (let [url (guisso-url service "/oauth2/token")
+         params (assoc params
+                       :client-id (:guisso-client-id service)
+                       :client-secret (:guisso-client-secret service))]
+     (-> (oauth/fetch-token url params)
+         (guisso-response->token-info)))
+   (catch [:status 400] {body :body}
```

Comment:
That would be `clj-http` used by `oauthentic`. See https://github.com/dakrone/clj-http#exceptions